### PR TITLE
NEW Log out all devices for members with no admin access

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-session-manager/issues/58

### CODE REVIEW

File: src/Security/LogOutAuthenticationHandler.php [hunk 1]
```diff
@@ -58,18 +70,60 @@ public function logOut(HTTPRequest $request = null)
...
+
+    /**
+     * Decides whether the provided user has access to any LeftAndMain controller, which indicates some level
+     * of access to the CMS.
+     *
+     * @see LeftAndMain::init()
+     * @param Member $member
+     * @return bool
+     */
+    private function hasAdminAccess(Member $member): bool
```
Comment (emteknetnz) [inline src/Security/LogOutAuthenticationHandler.php:101]: This function was copy pasted from https://github.com/silverstripe/silverstripe-mfa/blob/4/src/Service/EnforcementManager.php#L221

It's a little odd, though it works

It's a `private` function so easy to remove in the future if we add a function like this to core

**Review by maxime-rainville — COMMENTED** [review]
It's not clear to me that we should cater to users who do not have access to the CMS.

Most sites who allow users to register have very specific workflows in mind and will spent quite a bit of time thinking those through. If devs want to give allow users to log out from all devices, they can implement that feature in their front end.

In any case, this seems to be a completely different problem than what was initially raised in #46.

**Comment by emteknetnz** [comment]
Yeah agree it's pretty different and since it's adding new functionality so should probably split

Makes more sense to have team discussion on the parent issue rather than this pull-request - would you be able to move your comments to there?

**Comment by emteknetnz** [comment]
I've split the parent issue to a new issue https://github.com/silverstripe/silverstripe-session-manager/issues/58

**Comment by emteknetnz** [comment]
Closing this PR for now as it's very unlikely to be merged.  It can be reopened if there's a case for including it.

**Pull request closed without merging**